core/src/main/kotlin/io/specmatic/conversions/OpenApiSpecification.kt

```diff
@@ -1603,7 +1603,14 @@ class OpenApiSpecification(
                         discriminator = Discriminator.create(schema.discriminator?.propertyName, finalDiscriminatorMappings.keys.toSet(), finalDiscriminatorMappings)
                     )
                 } else if (schema.anyOf != null) {
-                    throw UnsupportedOperationException("Specmatic does not support anyOf")
+                    val candidatePatterns = schema.anyOf.map { componentSchema ->
+                        toSpecmaticPattern(componentSchema, typeStack, "")
```

Comment:
There should be a field passed to toSpecmaticPattern to default additionalProperties to FreeForm in JSONObjectPattern. The field should be false by default. It should be handled in OpenApiSpecification.toJSONObjectPattern (which means it will be passed from here to toSpecamticPattern, and from toSpecamticPattern to toJSONObjectPattern. And in toJSNObjectPattern, if this flag is true, and if schema.additionalProperties is null, the value of the JSONObjectPattern.additionalProperties property on the new object should be set to FreeForm.